### PR TITLE
:feat: Implement and update cat commands

### LINKED ISSUE #20: Implement cat command
- Overwrites the entire file

### PR DESCRIPTION
**Pull request by EeshaArif**
- Update rm command to remove data in sectors
- Update cat_end command for parsed entries
- Implement cat command

close #20

### CODE REVIEW

File: commands/cat_end.py [hunk 1]
```diff
@@ -1,37 +1,22 @@
...
+        path = self.arguments[0].data
+        file = self.context.parse(path, True)
+        file_directory = DirectoryDao.get_directory_from_directory_id(
+            file.directory_id)
 
-        if (FileDao.is_unique_filename(filename, current_directory)
-                or not FileDao.is_valid_filename(filename)):
+        if (FileDao.is_unique_filename(file.name, file_directory)
+                or not FileDao.is_valid_filename(file.name)):
             raise ValueError('InvalidEntry: File does not exist')
```
Comment (EeshaArif) [inline commands/cat_end.py:19]: Let me know if these checks are redundant and have been taken care of in the parse()

File: commands/cat_end.py [hunk 2]
```diff
@@ -1,37 +1,22 @@
...
-        filename = self.arguments[0].data
-        current_directory = self.context.current_directory
+        path = self.arguments[0].data
+        file = self.context.parse(path, True)
+        file_directory = DirectoryDao.get_directory_from_directory_id(
+            file.directory_id)
 
-        if (FileDao.is_unique_filename(filename, current_directory)
-                or not FileDao.is_valid_filename(filename)):
+        if (FileDao.is_unique_filename(file.name, file_directory)
```
Comment (EeshaArif) [inline commands/cat_end.py:17]: Let me know if these checks are redundant and have been taken care of in the parse()

File: commands/cat.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+class CatCommand(BaseCommand):
+    command = 'cat'
+    arguments = [StringArgument()]
+
+    def run(self):
+        path = self.arguments[0].data
+        file = self.context.parse(path, True)
+        file_directory = DirectoryDao.get_directory_from_directory_id(
+            file.directory_id)
```
Comment (Diaga) [inline commands/cat.py:15]: You can do `file.directory`

File: commands/cat.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+        if (FileDao.is_unique_filename(file.name, file_directory)
```
Comment (Diaga) [inline commands/cat.py:17]: We do not need these checks, `context.parse` will raise errors if the file does not exist.

File: commands/cat_end.py [hunk 5]
```diff
@@ -1,37 +1,22 @@
...
 class CatEndCommand(BaseCommand):
     command = 'cat_end'
     arguments = [StringArgument()]
 
     def run(self):
-        filename = self.arguments[0].data
-        current_directory = self.context.current_directory
+        path = self.arguments[0].data
+        file = self.context.parse(path, True)
+        file_directory = DirectoryDao.get_directory_from_directory_id(
```
Comment (Diaga) [inline commands/cat_end.py:14]: Same as above, `file.directory`

File: commands/cat_end.py [hunk 6]
```diff
@@ -1,37 +1,22 @@
...
+            file.directory_id)
 
-        if (FileDao.is_unique_filename(filename, current_directory)
-                or not FileDao.is_valid_filename(filename)):
+        if (FileDao.is_unique_filename(file.name, file_directory)
+                or not FileDao.is_valid_filename(file.name)):
```
Comment (Diaga) [inline commands/cat_end.py:18]: Same as above, we do not need these checks

**Review by Diaga — CHANGES_REQUESTED** [review]
Some small changes, looks good otherwise!

**Review by Diaga — APPROVED** [review]
Looks good, thanks~

**Pull request merged**